examples/pools/main.rs

```diff
@@ -61,40 +57,34 @@ fn main() {
         events::handle_instantiate_event,
     );
 
-    // Define a generic handler for transactions,
-    // which the processor will call for each transaction.
-    fn transaction_handler(
+    #[transaction_handler]
+    async fn transaction_handler(
         context: TransactionHandlerContext<AppState, TxHandle>,
     ) -> Result<(), TransactionHandlerError> {
-        // Do something like start a database transaction
-        let mut transaction_context =
-            context.app_state.async_runtime.block_on(async {
-                // start a database transaction
-                let tx = context.app_state.pool.begin().await.unwrap();
-                TxHandle { transaction: tx }
-            });
+        let tx = context.app_state.pool.begin().await.unwrap();
+        let mut transaction_context = TxHandle { transaction: tx };
 
         // Handle the events in the transaction
-        context.transaction.handle_events(
-            context.app_state,
-            context.handler_registry,
-            &mut transaction_context,
-        )?;
+        context
+            .incoming_transaction
+            .handle_events(
```

Comment:
I like the idea of abstracting away the loop 👍 . Maybe rename it to `process_events`?

Reply:
Not really a strong opinion on the naming here.

Reply:
Let's go with `process_events` then. We already have enough `hanlders`^^